core/src/main/java/org/elasticsearch/action/admin/cluster/snapshots/status/TransportSnapshotsStatusAction.java

```diff
@@ -197,20 +198,20 @@ private SnapshotsStatusResponse buildResponse(SnapshotsStatusRequest request, Li
         if (Strings.hasText(request.repository())) {
             if (request.snapshots() != null && request.snapshots().length > 0) {
                 for (String snapshotName : request.snapshots()) {
-                    SnapshotId snapshotId = new SnapshotId(request.repository(), snapshotName);
-                    if (currentSnapshotIds.contains(snapshotId)) {
+                    Snapshot snapshot = new Snapshot(request.repository(), snapshotName);
+                    if (currentSnapshotIds.contains(snapshot)) {
```

Comment:
This always returns false as `currentSnapshotIds` is of type `Set<SnapshotId>` and `snapshot` is of type `Snapshot`. Is this untested?